src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -145,19 +145,19 @@ public string HttpVersion
             {
                 if (_httpVersion == Http.HttpVersion.Http11)
                 {
-                    return HttpUtilities.Http11Version;
+                    return AspNetCore.Http.HttpProtocols.Http11;
                 }
                 if (_httpVersion == Http.HttpVersion.Http10)
                 {
-                    return HttpUtilities.Http10Version;
+                    return AspNetCore.Http.HttpProtocols.Http10;
                 }
```

Comment:
How often HTTP/1.0 is being used right now? Is it a good idea to put this check after HTTP/3?

Reply:
More than HTTP/3 😁.

Since these are direct version equality checks the ordering shouldn't matter much. We could run them in reverse (3,2,1,...) so our latest is always best.